paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -65,19 +71,36 @@ public BranchManager(
         this.schemaManager = schemaManager;
     }
 
+    /** Commit specify branch to main. */
+    public void commitMainBranch(String branchName) throws IOException {
+        Path mainBranchFile = new Path(tablePath, MAIN_BRANCH_FILE);
+        fileIO.delete(mainBranchFile, false);
+        fileIO.overwriteFileUtf8(mainBranchFile, branchName);
+    }
+
     /** Return the root Directory of branch. */
     public Path branchDirectory() {
         return new Path(tablePath + "/branch");
     }
 
     /** Return the path string of a branch. */
-    public static String getBranchPath(Path tablePath, String branchName) {
+    public static String getBranchPath(FileIO fileIO, Path tablePath, String branchName) {
+        if (StringUtils.isBlank(branchName)) {
```

Comment:
In what circumstances this branchName could be empty?

Reply:
> In what circumstances this branchName could be empty?

I have made some optimizations to the code, please review it again.